### PR TITLE
Add build rule for DAML Driver for PostgreSQL docs

### PR DESCRIPTION
**Pull request by bame-da**
CHANGELOG_BEGIN
CHANGELOG_END

This adds a build rule ``//ledger/daml-on-sql:docs`` for the docs of the DAML Driver for PostgreSQL using the DAML docs theme. This required some additional config options on the theme and fine-tuning of the version switcher to be able to have independent versions for main docs and driver docs.

The end-result can be seen on https://staging-docs.daml.com/ and https://staging-docs.daml.com/daml-driver-for-postgresql/index.html respectively.

A followup PR will be needed to release this to docs.daml.com.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: docs/theme/js/switcher.js [hunk 1]
```diff
@@ -87,10 +87,36 @@
       return ''
     }
   
+    function nthIndex(str, pat, n){
+      var L= str.length, i= -1;
```
Comment (cocreature) [inline docs/theme/js/switcher.js:91]: Looks like you copied those functions from stackoverflow but imho that’s pretty unreadable. Iterating over regexp matches which already support that natively here and just calling `.length` on the result for `count` would be much cleaner imho.

File: docs/theme/js/switcher.js [hunk 2]
```diff
@@ -87,10 +87,36 @@
...
+      return n;
+    }
+
     function versions_root_in_url(url) {
-      var version = version_segment_in_url(url)
-      if(version == '') return get_host()
-      else return url.substring(0, url.indexOf(version))
+      var version = version_segment_in_url(url);
+      if(version == '') {
+        var depth_from_root = count(DOCUMENTATION_OPTIONS.URL_ROOT, "../");
```
Comment (cocreature) [inline docs/theme/js/switcher.js:113]: What exactly results in this being set differently for the daml on sql docs? I couldn’t find anything that could result in sphinx adding a different path here.

**Review by cocreature — APPROVED** [review]
LGTM with a few minor comments, thanks!

File: docs/theme/docs/conf.py [hunk 3]
```diff
@@ -97,7 +97,9 @@
 # further.  For a list of options available for each theme, see the
 # documentation.
 html_theme_options = {
```
[the remark this answers is not in the document]
Comment (bame-da) [reply docs/theme/docs/conf.py:99]: `theme/docs/da_theme_skeleton/theme.conf` is a part of the Sphinx theme. It specifies what theme options are available in the theme, ie what can go into the `html_theme_options` block in `conf.py`.

`docs/theme/docs/conf.py` is used for the theme preview, but I'll have a go at getting rid of that.

`docs/configs/html/conf.py` is the config for our actual docs build.

File: docs/theme/js/switcher.js (hunk 1 above)
Comment (bame-da) [reply docs/theme/js/switcher.js:91]: How would you implement `nthIndex` using RegExp matches? I can see it for `count`:

```
  function count(str, pat){
      var re = new RegExp(pat, "g");
      return (str.match(re) || []).length;
    }
```

Comment (cocreature) [reply docs/theme/js/switcher.js:91]: Iterate `n` times over the result of `exec`

File: docs/theme/js/switcher.js (hunk 2 above)
Comment (bame-da) [reply docs/theme/js/switcher.js:113]: This change is about being able to version the two docs independently. The old script first looked for a version string in the path so if you were at `https://example.com/foo/1.0.0/bar/baz.html`, it knew to look for `https://example.com/foo/versions.json` and replace the `1.0.0` with the selected version. As a fallback it looked for `https://example.com/versions.json`

The trouble now is that now you may be at `https://docs.daml.com/daml-driver-for-postgres/foo/bar.html` and need to figure out that your "root" is actually `https://docs.daml.com/daml-driver-for-postgres/`. Our theme already sets the `DOCUMENTATION_OPTIONS.URL_ROOT` property which is the relative path to the root of the sphinx tree. Afaict it's always `./` or `../` repeated. I can't use `https://docs.daml.com/daml-driver-for-postgres/foo/../` as the root, as that leads to bugs when trying to match on `root/version`. Ie I end up trying to match `https://docs.daml.com/daml-driver-for-postgres/foo/../` with `https://docs.daml.com/daml-driver-for-postgres/` which fails so I decided to just count depth and "navigate up".

Comment (cocreature) [reply docs/theme/js/switcher.js:113]: I get the change, what I don’t get is how `URL_ROOT` is set. The readme seems to be copied to `source/index.rst` which seems like it should be `./` not `../`.

File: docs/theme/js/switcher.js (hunk 1 above)
Comment (bame-da) [reply docs/theme/js/switcher.js:91]: Interesting, I didn't know about the stateful use of `exec`. This is what I came up with. Is it easier to understand in your opinion?

```
    function nthIndex(str, pat, n){
      var re = new RegExp(pat, "g");
      var m = null;
      for(var i = 0; i < n; i++)
        m = re.exec(str);
      return m ? re.lastIndex : -1;
    }
```

Comment (cocreature) [reply docs/theme/js/switcher.js:91]: I like it, thx!

File: docs/theme/js/switcher.js (hunk 2 above)
Comment (bame-da) [reply docs/theme/js/switcher.js:106]: Oh, that's set through Sphinx's template engine: https://github.com/digital-asset/daml/pull/8188/files#diff-d4ebd498d60518b23a517eb9e53ca719f9536768a6d9eaa6fda1b314bf986807L48-L58

It's always the relative path to `/` from Sphinx's perspective, which is the `source` folder.

Comment (bame-da) [reply docs/theme/js/switcher.js:106]: And indeed, it's set to `./` in `index.html`.

**Pull request merged**